subsys/net/ip/net_context.c

```diff
@@ -689,6 +689,45 @@ int net_context_put(struct net_context *context)
 	return 0;
 }
 
+static int net_context_bind_default(struct net_context *context,
+				    const struct sockaddr *dest_addr)
+{
+	struct net_if *iface = net_if_get_default();
+
+	if (!iface) {
+		NET_ERR("Cannot bind to %s",
+			net_sprint_ipv6_addr(&addr6->sin6_addr));
+
+		return -EADDRNOTAVAIL;
```

Comment:
Sorry, this code copied from net_context_bind(). If you think checking needs to be done differently, then net_context_bind() needs to do it, but I don't see anything wrong with what it does. Again, bind_default() function performs the same operation as bind(.., INADDR_ANY, ...), the reason I don't use bind() directly is because it requires munging with address families to use particular INADDR_ANY impl for each. (bind_default also checks before binds.)